### PR TITLE
Right click node

### PR DESCRIPTION
**Pull request by rnugraha**
Bing-bang pull request, consists of:
- Metadata refactoring
- Tree refactoring
- Unit testings for related services and components

### CODE REVIEW

**Comment by berndvdveen** [comment]
What a monster! Will review tomorrow

File: src/app/components/study-accordion/study-accordion.directive.js [hunk 1]
```diff
@@ -1,194 +1,66 @@
...
-      /**
-       * When a node is clicked, get children of the children
-       * @param node
-       */
-      scope.populateChilds = function (node) {
-        node.nodes.forEach(function(child){
-          _getNodeChildren(child, false, '');
-        });
-      };
+      $scope.populateChilds = function (node) {
```
Comment (kretep) [inline src/app/components/study-accordion/study-accordion.directive.js]: Childs? You mean children?

File: src/app/components/study-accordion/study-accordion.tpl.html [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+  <table style="font-size: .9em" click-outside="closeThis()">
+    <tr>
+      <td><b>Fullname:</b></td>
+      <td>&nbsp;&nbsp;</td>
+      <td>{{metadataObj.fullname}}</td>
+    </tr>
+    <tr ng-repeat="(key, value) in metadataObj.body" class="popover-table-row"
+        ng-class-even="popover-even-row">
+      <td><b>{{key}}:</b></td>
+      <td>&nbsp;&nbsp;</td>
```
Comment (kretep) [inline src/app/components/study-accordion/study-accordion.tpl.html]: Maybe use CSS for this?

**Comment by kretep** [comment]
Looks ok. What does @berndvdveen think?

**Pull request merged**